config_converter/config_mapper/config_mapper.py

```diff
@@ -0,0 +1,189 @@
+"""Program that converts a fluentd config file to a master agent config file.
+
+Usage: Run the exec file in the bin folder with 2 arguments - the
+path to the config file to migrate, and a directory to store the output
+master agent config file in.
+
+To do:
+1. Improve the CLI. Include a --help option. Allow optional args like
+   log-level and log-filepath (see project docs).
+2. Create a mapping for log_level if there in the fluentd config file and input
+   default ones if not given
+3. Generate stats like #fields converted, ignored, and output as schema
+4. Generate logs
+5. Fix types of numbers, lists
+6. Generate tests to check if for right logs, right stats, above changes
+"""
+
+import copy
+import sys
+import yaml
+from google.protobuf import json_format
+from config_converter.config_mapper import config_pb2
+
+# each sub dict contains fields that stay in the current level only
+_MAIN_MAP = {
+    'in_tail': {
+        'exclude_path': 'exclude_path',
+        'path': 'path',
+        'path_key': 'path_field_name',
+        'pos_file': 'checkpoint_file',
+        'refresh_interval': 'refresh_interval',
+        'rotate_wait': 'rotate_wait'
+    }
+}
+# each sub list contains fields that need to be treated differently
+# https://docs.fluentd.org/parser/multiline - shows formatN works for
+# 1 <= N <= 20
+_SPECIAL_MAP = {
+    'in_tail': [
+        'format', 'format_firstline', '@type', 'multiline_flush_interval',
+        'expression'
+    ] + [f'format{i}' for i in range(1, 21)]
+}
+# dictionary of dictionaries - one per main plugin type (source, match)
+# each dict contains fields that go in the upper level
+_OUTER_MAP = {'source': {'tag': 'name', '@type': 'type'}}
+# each sub list contains known and allowed sub directives
+_SUPPORTED_DIRS = {'in_tail': ['parse']}
```

Comment:
I think explaining the meaning of the key would make it more clear.

Reply:
Good point - I will update the comments for all the global constants to be more clear.